wsmaster/che-core-api-workspace-activity/src/main/java/org/eclipse/che/api/workspace/activity/JpaWorkspaceActivityDao.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.activity;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.inject.persist.Transactional;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import javax.persistence.EntityManager;
+import org.eclipse.che.api.core.ServerException;
+
+/**
+ * JPA workspaces expiration times storage.
+ *
+ * @author Max Shaposhnik (mshaposh@redhat.com)
+ */
+@Singleton
+public class JpaWorkspaceActivityDao implements WorkspaceActivityDao {
```

Comment:
I don't see Tck tests for it. Is there any plans to do them? Please don't forget to add them to postgresql integration tests.

Reply:
Added